gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java

```diff
@@ -87,8 +89,14 @@ public ReferralRequest mapToReferralRequest(RCMRMT030101UKEhrExtract ehrExtract,
         var authoredOn = getAuthoredOn(requestStatement.getAvailabilityTime());
         var referralPriority = getReferralPriority(requestStatement);
 
+        final Meta meta = confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+            META_PROFILE,
+            requestStatement.getConfidentialityCode(),
+            ehrComposition.getConfidentialityCode()
+        );
+
         referralRequest.setId(id);
-        referralRequest.setMeta(generateMeta(META_PROFILE));
+        referralRequest.setMeta(meta);
```

Comment:
not directly related to your changes however I think you can take advantage of the Fluent API here and remove all those duplications referring to referralRequest object.

Reply:
Hey Ole, great point! I have attempted this, it doesn't seem to want to play nicely in terms of moving it to a fluent API.

```java
referralRequest
    .setId(id)
    .setMeta(meta)
    .getIdentifier().add(identifier); // Cannot resolve method 'getIdentifier' in 'Resource'
```